web/backend/app.js

```diff
@@ -20,29 +23,56 @@ if (process.env.HTTPS_CERT_PATH && process.env.HTTPS_KEY_PATH) {
   server = new http.Server(app);
 }
 
-// Connect to the UNIX-socket
-const UNIX_client_socket = net.createConnection(process.env.UNIX_SOCKET_PATH || "/tmp/thi_drone");
-
-// Start to listen with server
-const port = process.env.PORT || 3000;
-server.listen(port, () => {
-  console.log(`Server is running on port ${port}`);
-});
-
-// Initiate web-socket on server
-const wss = new websocket.server({
-  httpServer: server,
-  autoAcceptConnections: false,
-  keepaliveInterval: 5000,
-});
-
-// Accept connections on "drone-logging"
-wss.on("request", async function (request) {
-  request.accept("drone-logging", request.origin);
-});
-
-// Broadcast all data received on the UNIX-socket to all web-socket clients
-UNIX_client_socket.on("data", (data) => {
-  console.log("Sending", data.toString(), "to all clients.");
-  wss.broadcast(data);
-});
+// Function to wait for the UNIX socket file to exist
+const waitForSocket = (socketPath, interval) => {
+  return new Promise((resolve) => {
+    const checkSocket = () => {
+      if (fs.existsSync(socketPath)) {
+        resolve();
+      } else {
+        console.log(`Waiting for socket ${socketPath} to exist...`);
+        setTimeout(checkSocket, interval);
+      }
+    };
+    checkSocket();
+  });
+};
+
+// Main function to start the server and handle connections
+const startServer = async () => {
+  const UNIX_SOCKET_PATH = process.env.UNIX_SOCKET_PATH || "/tmp/thi_drone";
+  const CHECK_INTERVAL = 1000;
+
+  await waitForSocket(UNIX_SOCKET_PATH, CHECK_INTERVAL);
+
+  console.log(`Socket ${UNIX_SOCKET_PATH} exists. Proceeding to connect...`);
+
+  // Connect to the UNIX-socket
+  const UNIX_client_socket = net.createConnection(UNIX_SOCKET_PATH);
```

Comment:
What happens if the socket gets closed right after the check?

Reply:
Yeah you're right. The telemetry_node would restart, and I implemented (29eff01)[https://github.com/THI-Drone/Ground_Station/pull/6/commits/29eff01d900fda7aa604a409e6299319418ff2a1]